crates/iceberg-catalog/src/service/storage/az.rs

```diff
@@ -0,0 +1,536 @@
+            .map_err(|err| {
+                ErrorModel::precondition_failed(
+                    "Failed to delete test location.",
+                    "AzStorageDeletionError",
+                    Some(Box::new(err)),
+                )
+            })?;
+
+        tracing::debug!("Cleaned up");
+
+        Ok(())
+    }
+
+    /// Check if the profile can be updated with the other profile.
+    /// `key_prefix`, `region` and `bucket` must be the same.
+    /// We enforce this to avoid issues by accidentally changing the bucket or region
+    /// of a warehouse, after which all tables would not be accessible anymore.
+    /// Changing an endpoint might still result in an invalid profile, but we allow it.
+    ///
+    /// # Errors
+    /// Fails if the `bucket`, `region` or `key_prefix` is different.
+    pub fn can_be_updated_with(&self, other: &Self) -> Result<()> {
+        if self.filesystem != other.filesystem {
+            return Err(ErrorModel::bad_request(
+                "Storage Profile `filesystem` cannot be updated to prevent data loss.",
+                "InvalidFileSystem",
+                None,
+            )
+            .into());
+        }
+
+        if self.key_prefix != other.key_prefix {
+            return Err(ErrorModel::bad_request(
+                "Storage Profile `key_prefix` cannot be updated to prevent data loss.",
+                "InvalidKeyPrefix",
+                None,
+            )
+            .into());
+        }
+
+        Ok(())
+    }
+
+    // may change..
+    #[allow(clippy::unused_self)]
+    #[must_use]
+    pub fn generate_catalog_config(&self, _: WarehouseIdent) -> CatalogConfig {
+        CatalogConfig {
+            defaults: HashMap::default(),
+            overrides: HashMap::default(),
+        }
+    }
+
+    #[must_use]
+    pub fn initial_tabular_location(
+        &self,
+        namespace_id: NamespaceIdentUuid,
+        tabular_id: TabularIdentUuid,
+    ) -> String {
+        format!(
+            "{}{tabular_id}",
+            self.initial_namespace_location(namespace_id)
+        )
+    }
+
+    fn initial_namespace_location(&self, namespace_id: NamespaceIdentUuid) -> String {
+        format!("{}{namespace_id}/", self.location())
+    }
+
+    fn location(&self) -> String {
+        if let Some(key_prefix) = &self.key_prefix {
+            format!(
+                "abfss://{}@{}.{}/{}/",
+                self.filesystem,
+                self.account_name,
+                self.endpoint_suffix
+                    .as_deref()
+                    .unwrap_or(DEFAULT_ENDPOINT_SUFFIX),
+                key_prefix.trim_matches('/')
+            )
+        } else {
+            format!(
+                "abfss:/{}@{}.{}/",
+                self.filesystem,
+                self.account_name,
+                self.endpoint_suffix
+                    .as_deref()
+                    .unwrap_or(DEFAULT_ENDPOINT_SUFFIX)
+                    .trim_end_matches('/'),
```

Comment:
I am not sure if this could be used maliciously, but maybe we should make sure there are no "/" altogether in the account_name and endpoint_suffix?